java/developer-guide/working-with-outlook-items/working-with-recurrences/_index.md

```diff
@@ -0,0 +1,527 @@
+### **Daily Recurrences: EndAfterNOccurrence Recurrance Type**
+In this type of recurrence, number of recurrences is to be set along with other information as follows:
+
+1. Set start, end and due date.
+1. Create a MapiTask.
+1. Set task state to NotAssigned.
+1. Create the daily recurrence object by setting the properties like PatternType, Period, WeekStartDay, EndType and OccurenceCount.
+1. Set MapiTask.Recurrence property to this daily recurrence object.
+1. Save this message on disc.
+
+The following code snippet shows you how to create task with recurrence end type as EndAfterNOccurrence.
+
+
+
+~~~Java
+// For complete examples and data files, please go to https://github.com/aspose-email/Aspose.Email-for-Java
+// The path to the File directory.
+String dataDir = RunExamples.getDataDir_Outlook();
+
+Date startDate = getDate(2015, 7, 16);
+Date dueDate = getDate(2015, 7, 16);
+Date endByDate = getDate(2015, 8, 1);
+ 
+MapiTask task = new MapiTask("This is test task", "Sample Body", startDate, dueDate);
+task.setState(MapiTaskState.NotAssigned);
+
+// Set the Daily recurrence
+MapiCalendarDailyRecurrencePattern rec = new MapiCalendarDailyRecurrencePattern();
+rec.setPatternType(MapiCalendarRecurrencePatternType.Day);
+rec.setPeriod(1);
+rec.setWeekStartDay(DayOfWeek.Sunday);
+rec.setEndType(MapiCalendarRecurrenceEndType.EndAfterNOccurrences);
+rec.setOccurrenceCount(getOccurrenceCount(startDate, endByDate, "FREQ=DAILY"));
+
+if (rec.getOccurrenceCount() == 0)
+{
+    rec.setOccurrenceCount(1);
+}
+
+task.setRecurrence(rec);
+task.save(dataDir + "Daily_out.msg", TaskSaveFormat.Msg);
+~~~
+
+
+Following function can be used to calculate the number of events between the two dates:
+
+
+
+~~~Java
+private static long getOccurrenceCount(Date start, Date endBy, String rrule)
+{
+    DateFormat formatter = new SimpleDateFormat("yyyyMMdd");
+    CalendarRecurrence pattern = new CalendarRecurrence("DTSTART:" + formatter.format(start) + "\r\nRRULE:" + rrule);
+    DateCollection dates = pattern.generateOccurrences(start, endBy);
+    return dates.size();
+}
+~~~
+#### **Setting the Occurrences count value**
+The following code snippet shows you how to set the occurrences count value.
+
+
+
+~~~Java
+// For complete examples and data files, please go to https://github.com/aspose-email/Aspose.Email-for-Java
+// Set the Daily recurrence
+MapiCalendarDailyRecurrencePattern record = new MapiCalendarDailyRecurrencePattern();
+
+record.setPatternType(MapiCalendarRecurrencePatternType.Day);
+record.setPeriod(1);
+record.setWeekStartDay(DayOfWeek.Sunday);
+record.setEndType(MapiCalendarRecurrenceEndType.EndAfterNOccurrences);
+record.setOccurrenceCount(5);
+task.setRecurrence(record);
+~~~
+### **Daily Recurrences: EndAfterDate Recurrance Type**
+"End By" option in the Mapi Task is achieved by setting the OccurrenceCount property calculated by the getOccurrenceCount() function. This function takes start date , end date and RRULE string.
```

Comment:
`"End By" option in the Mapi Task is achieved by setting the *OccurrenceCount* property calculated by the **getOccurrenceCount()** function. This function takes start date, end date and RRULE string.`